### DISCUSSION TITLE
Failed to restart lvm2-lvmpolld.service: Unit lvm2-lvmpolld.socket is masked.

### DISCUSSION THREAD

**Discussion by Coolkid2342** [cockpit-project/cockpit#16042] [Q&A] [answered]
When trying to install cockpit-machines it keeps throwing a error of: 

Failed to restart lvm2-lvmpolld.service: Unit lvm2-lvmpolld.socket is masked.
invoke-rc.d: initscript lvm2-lvmpolld, action "restart" failed.
* lvm2-lvmpolld.service - LVM2 poll daemon
     Loaded: loaded (/lib/systemd/system/lvm2-lvmpolld.service; static; vendor preset: enabled)
     Active: inactive (dead)
       Docs: man:lvmpolld(8)
dpkg: error processing package lvm2 (--configure):
 installed lvm2 package post-installation script subprocess returned error exit status 1
Setting up humanity-icon-theme (0.6.15) ...
Setting up libvirt-daemon-system (6.0.0-0ubuntu8.10) ...
A dependency job for libvirtd.service failed. See 'journalctl -xe' for details.
Job failed. See "journalctl -xe" for details.
Job failed. See "journalctl -xe" for details.
virtlockd.service is a disabled or a static unit, not starting it.
virtlogd.service is a disabled or a static unit, not starting it.

Causing me to be unable to create virtual machines or create virtual networks inside of cockpit.

**Comment by martinpitt** [comment, marked as answer]
This is a local configuration error -- Apparently you masked that unit. It's unrelated to cockpit or cockpit-machines.

You sohuld be able to fix it with

    sudo systemctl unmask lvm2-lvmpolld.service
    sudo systemctl restart lvm2-lvmpolld libvirtd

**Reply by Coolkid2342** [reply]
Not sure how it became masked since just ran the `apt install cockpit` and `apt install cockpit-machines`. Running the unmask command and restarting the services just returns back with the service is masked again. gonna try a fresh install just incase another program is causing this error

**Reply by martinpitt** [reply]
Certainly the mere installation of cockpit doesn't cause that -- the lvm service must have been broken before already, it's just that cockpit-machines depends on libvirt package, which tried to start libvirtd.service, which fails on the already broken lvm. I.e. this merely made the error visible. 

It may also be that you have some removed, but not purged package? `invoke-rc.d` sounds like Debian/Ubuntu at least, what does `dpkg -s $(dpkg -S /lib/systemd/system/lvm2-lvmpolld.service)` show?

**Reply by Coolkid2342** [reply]
weird then. the results show `dpkg-query: error: --status needs a valid package name but 'lvm2:' is not: illegal architecture name in specifier 'lvm2:': may not be empty string` currently in the process to do a fresh install of ubuntu server to see if it just fixes itself

**Reply by martinpitt** [reply]
Argh, sorry about that. The result of the `dpkg -S` lookup was `lvm2`, so the status of that package would be `dpkg -s lvm2`. That may be "removed" or maybe in some inconsistent failure state, instead of "installed" or "not present"

**Reply by Coolkid2342** [reply]
after a fresh install its seems to be working fine so seems like something was locking itself to lvm2 and not letting and changes
